catalogd/internal/controllers/core/clustercatalog_controller.go

```diff
@@ -216,50 +222,58 @@ func (r *ClusterCatalogReconciler) reconcile(ctx context.Context, catalog *catal
 	case catalog.Generation != storedCatalog.observedGeneration:
 		l.Info("unpack required: catalog generation differs from observed generation")
 		needsUnpack = true
-	case r.needsPoll(storedCatalog.unpackResult.LastSuccessfulPollAttempt.Time, catalog):
+	case r.needsPoll(storedCatalog.lastSuccessfulPoll, catalog):
 		l.Info("unpack required: poll duration has elapsed")
 		needsUnpack = true
 	}
 
 	if !needsUnpack {
 		// No need to update the status because we've already checked
 		// that it is set correctly. Otherwise, we'd be unpacking again.
-		return nextPollResult(storedCatalog.unpackResult.LastSuccessfulPollAttempt.Time, catalog), nil
+		return nextPollResult(storedCatalog.lastSuccessfulPoll, catalog), nil
+	}
+
+	if catalog.Spec.Source.Type != catalogdv1.SourceTypeImage {
+		err := reconcile.TerminalError(fmt.Errorf("unknown source type %q", catalog.Spec.Source.Type))
+		updateStatusProgressing(&catalog.Status, catalog.GetGeneration(), err)
+		return ctrl.Result{}, err
+	}
```

Comment:
So we never checked the `catalog.Spec.Source.Type` or did we move this from other place?

Reply:
That used to be part of the `source.ContainersImage` unpacker code: https://github.com/operator-framework/operator-controller/blob/becde514411344e71d51e0fcb8ea8a840e6fb1a1/internal/catalogd/source/containers_image.go#L43-L49